From 43db1297e8798cab5971ad6665306b1b87823a21 Mon Sep 17 00:00:00 2001
From: Christophe Roullier <christophe.roullier@foss.st.com>
Date: Wed, 17 Jan 2024 09:41:08 +0100
Subject: [PATCH] dts: arm64: st: add wakeup source for Ethernets for stm32mp25

When the "wakeup-source" flag is found, Ethernet marks
the device as wakeup capable and enables the wakeup by
default. (with WoL event)

Signed-off-by: Christophe Roullier <christophe.roullier@foss.st.com>

Change-Id: I1757f79041c9967bebe87a67b97536a6c92c79fb
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/354888
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Tested-by: Christophe ROULLIER <christophe.roullier@st.com>
Reviewed-by: Christophe ROULLIER <christophe.roullier@st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Domain-Review: Christophe ROULLIER <christophe.roullier@st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/386400
Reviewed-by: Christophe ROULLIER <christophe.roullier@foss.st.com>
Tested-by: Christophe ROULLIER <christophe.roullier@foss.st.com>
Domain-Review: Christophe ROULLIER <christophe.roullier@foss.st.com>
---
 arch/arm64/boot/dts/st/stm32mp251.dtsi | 7 +++++--
 arch/arm64/boot/dts/st/stm32mp253.dtsi | 7 +++++--
 2 files changed, 10 insertions(+), 4 deletions(-)

--- a/arch/arm64/boot/dts/st/stm32mp251.dtsi
+++ b/arch/arm64/boot/dts/st/stm32mp251.dtsi
@@ -2282,8 +2282,10 @@
 				compatible = "st,stm32mp25-dwmac", "snps,dwmac-5.10a";
 				reg = <0x482c0000 0x4000>;
 				reg-names = "stmmaceth";
-				interrupts-extended = <&intc GIC_SPI 130 IRQ_TYPE_LEVEL_HIGH>;
-				interrupt-names = "macirq";
+				interrupts-extended = <&intc GIC_SPI 130 IRQ_TYPE_LEVEL_HIGH>,
+						      <&exti1 68 IRQ_TYPE_LEVEL_HIGH>;
+				interrupt-names = "macirq",
+						  "eth_wake_irq";
 				clock-names = "stmmaceth",
 					      "mac-clk-tx",
 					      "mac-clk-rx",
@@ -2303,6 +2305,7 @@
 				snps,tso;
 				access-controllers = <&rifsc 60>;
 				power-domains = <&CLUSTER_PD>;
+				wakeup-source;
 				status = "disabled";
 				snps,mtl-rx-config = <&mtl_rx_setup_1>;
 				snps,mtl-tx-config = <&mtl_tx_setup_1>;
--- a/arch/arm64/boot/dts/st/stm32mp253.dtsi
+++ b/arch/arm64/boot/dts/st/stm32mp253.dtsi
@@ -134,8 +134,10 @@
 		compatible = "st,stm32mp25-dwmac", "snps,dwmac-5.10a";
 		reg = <0x482d0000 0x4000>;
 		reg-names = "stmmaceth";
-		interrupts-extended = <&intc GIC_SPI 133 IRQ_TYPE_LEVEL_HIGH>;
-		interrupt-names = "macirq";
+		interrupts-extended = <&intc GIC_SPI 133 IRQ_TYPE_LEVEL_HIGH>,
+				      <&exti1 70 IRQ_TYPE_LEVEL_HIGH>;
+		interrupt-names = "macirq",
+				  "eth_wake_irq";
 		clock-names = "stmmaceth",
 			      "mac-clk-tx",
 			      "mac-clk-rx",
@@ -155,6 +157,7 @@
 		snps,tso;
 		access-controllers = <&rifsc 61>;
 		power-domains = <&CLUSTER_PD>;
+		wakeup-source;
 		status = "disabled";
 		snps,mtl-rx-config = <&mtl_rx_setup_2>;
 		snps,mtl-tx-config = <&mtl_tx_setup_2>;
